### PR TITLE
MES-6794 Cat D State

### PR DESCRIPTION
**Pull request by Cullima**
## Description
- Adds state for Cat D

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/providers/category-whitelist/category-whitelist.ts [hunk 1]
```diff
@@ -11,6 +11,9 @@ export const whiteList : TestCategory[] = [
   TestCategory.C1E,
   TestCategory.CCPC,
   TestCategory.DCPC,
+  TestCategory.D,
+  TestCategory.D1,
+  TestCategory.D1E,
```
Comment (matthew2564) [inline src/app/providers/category-whitelist/category-whitelist.ts:16]: missing TestCategory.DE

File: src/store/tests/journal-data/cat-d/candidate/candidate.cat-d.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  previousADITests: null,
+  ethnicityCode: null,
+  businessAddress: {},
+  businessName: null,
+  businessTelephone: null,
+};
+
+export const candidateCatDReducer = createReducer(
+  initialState,
+  on(candidateActions.PopulateCandidateDetailsCatD, (state, { candidate }) => candidate),
```
Comment (matthew2564) [inline src/store/tests/journal-data/cat-d/candidate/candidate.cat-d.reducer.ts]: if we dont need to use the state object itself, the common approach I've seen is to just use an underscore i.e.

on(candidateActions.PopulateCandidateDetailsCatD, (_, { candidate }) => candidate),

File: src/store/tests/test-data/cat-d/safety-questions/safety-questions.cat-d.action.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { createAction } from '@ngrx/store';
+import { QuestionOutcome } from '@dvsa/mes-test-schema/categories/common';
+
+export const SAFETY_QUESTION_OUTCOME_CHANGED = '[VehicleChecksPage] [CatD] Safety Question Outcome Changed';
+export const ADD_SAFETY_QUESTION_COMMENT = '[Vehicle Checks] [CatD] Add SafetyQuestion comment';
```
Comment (matthew2564) [inline src/store/tests/test-data/cat-d/safety-questions/safety-questions.cat-d.action.ts]: are we able to remove the consts here?

File: src/store/tests/tests-reducer-factory.ts [hunk 4]
```diff
@@ -38,6 +40,10 @@ export function testsReducerFactory(
     case TestCategory.CCPC:
     case TestCategory.DCPC:
       return testsCatCPCReducer(action, state as Required<TestResultCatCPCSchema>);
+    case TestCategory.D:
+    case TestCategory.DE:
+    case TestCategory.D1E:
```
Comment (matthew2564) [inline src/store/tests/tests-reducer-factory.ts:45]: missing 

case TestCategory.D1:

File: src/store/tests/tests-reducer-factory.ts [hunk 5]
```diff
@@ -10,6 +10,8 @@ import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/
 import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
 
 import { testsCatBReducer } from './tests.cat-b.reducer';
+import { testsCatDReducer } from '@store/tests/tests.cat-d.reducer';
+import { CatDUniqueTypes } from '@dvsa/mes-test-schema/categories/D';
```
Comment (matthew2564) [inline src/store/tests/tests-reducer-factory.ts:13]: this should error with eslint due to path imports needing to be ordered

File: src/app/providers/category-whitelist/category-whitelist.ts (hunk 1 above)
Comment (Cullima) [reply src/app/providers/category-whitelist/category-whitelist.ts:16]: thanks. added

File: src/store/tests/journal-data/cat-d/candidate/candidate.cat-d.reducer.ts (hunk 2 above)
Comment (Cullima) [reply src/store/tests/journal-data/cat-d/candidate/candidate.cat-d.reducer.ts]: updated

File: src/store/tests/test-data/cat-d/safety-questions/safety-questions.cat-d.action.ts (hunk 3 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-d/safety-questions/safety-questions.cat-d.action.ts]: removed

File: src/store/tests/tests-reducer-factory.ts (hunk 4 above)
Comment (Cullima) [reply src/store/tests/tests-reducer-factory.ts:45]: added

File: src/store/tests/tests-reducer-factory.ts (hunk 5 above)
Comment (Cullima) [reply src/store/tests/tests-reducer-factory.ts:13]: updated thanks

**Review by matthew2564 — APPROVED** [review]

**Review by tomcrawleyy — APPROVED** [review]

**Pull request merged**